lib/matplotlib/tests/test_determinism.py

```diff
@@ -0,0 +1,153 @@
+    fig.savefig(stdout, format=fmt)
+
+
+@pytest.mark.parametrize(
+    "objects, fmt, usetex", [
+        ("", "pdf", False),
+        ("m", "pdf", False),
+        ("h", "pdf", False),
+        ("i", "pdf", False),
+        ("mhi", "pdf", False),
+        ("mhi", "ps", False),
+        pytest.param(
+            "mhi", "ps", True, marks=[needs_usetex, needs_ghostscript]),
+        ("mhi", "svg", False),
+        pytest.param("mhi", "svg", True, marks=needs_usetex),
+    ]
+)
+def test_determinism_check(objects, fmt, usetex):
+    """
+    Output three times the same graphs and checks that the outputs are exactly
+    the same.
+
+    Parameters
+    ----------
+    objects : str
+        Objects to be included in the test document: 'm' for markers, 'h' for
+        hatch patterns, 'i' for images.
+    fmt : {"pdf", "ps", "svg"}
+        Output format.
+    """
+    plots = [
+        subprocess.check_output(
+            [sys.executable, "-R", "-c",
+             f"from matplotlib.tests.test_determinism import _save_figure;"
+             f"_save_figure({objects!r}, {fmt!r}, {usetex})"],
+            env={**os.environ, "SOURCE_DATE_EPOCH": "946684800"})
+        for _ in range(3)
+    ]
+    for p in plots[1:]:
+        if fmt == "ps" and usetex:
+            if p != plots[0]:
+                pytest.skip("failed, maybe due to ghostscript timestamps")
+        else:
+            assert p == plots[0]
+
+
+@pytest.mark.parametrize(
+    "fmt, string", [
+        ("pdf", b"/CreationDate (D:20000101000000Z)"),
+        # SOURCE_DATE_EPOCH support is not tested with text.usetex,
+        # because the produced timestamp comes from ghostscript:
+        # %%CreationDate: D:20000101000000Z00\'00\', and this could change
+        # with another ghostscript version.
+        ("ps", b"%%CreationDate: Sat Jan 01 00:00:00 2000"),
+    ]
+)
+def test_determinism_source_date_epoch(fmt, string, keyword=b"CreationDate"):
+    """
+    Test SOURCE_DATE_EPOCH support. Output a document with the environment
+    variable SOURCE_DATE_EPOCH set to 2000-01-01 00:00 UTC and check that the
+    document contains the timestamp that corresponds to this date (given as an
+    argument).
+
+    Parameters
+    ----------
+    fmt : {"pdf", "ps", "svg"}
+        Output format.
+    string : str
+        Timestamp string for 2000-01-01 00:00 UTC.
+    keyword : bytes
+        A string to look at when searching for the timestamp in the document
+        (used in case the test fails).
+    """
+    buff = subprocess.check_output(
+        [sys.executable, "-R", "-c",
+         f"from matplotlib.tests.test_determinism import _save_figure; "
+         f"_save_figure('', {fmt!r})"],
+        env={**os.environ, "SOURCE_DATE_EPOCH": "946684800"})
+    find_keyword = re.compile(b".*" + keyword + b".*")
+    key = find_keyword.search(buff)
+    if key:
+        print(key.group())
+    else:
+        print("Timestamp keyword (%s) not found!" % keyword)
```

Comment:
f-string, too?

Reply:
I decided to just delete that chunk, which is anyways only involved in formatting the message on test failure.